### ISSUE TITLE
Pipe not working with Stereo Tool

### ISSUE THREAD

**Issue by RM-FM** [savonet/liquidsoap#883]
Basically, it seems quite easy to use Stereo Tool with Liquidsoap. We managed it to put it into a pipe using stdin/stdout. Unfortunately we do not hear any audio (just silence) as soon as we enable the pipe/process line in the script. Everything is working fine without.

According to the log, it looks like the process just stops stdin for some reason. Stereo Tool does not report any errors,  but the pipe process reports „stdin closed“ and Stereo Tool and Liquidsoap just freezes afree processing of 60K of data.

Please find a snippet of our code as well as the log below...

```
radio = cue_cut(request.dynamic(get_request))
radio = crossfade(radio)
radio = mksafe(radio)

processed = pipe(process='/etc/liquidsoap/stereo_tool_cmd_64 - - -s /etc/liquidsoap/rm.sts -k "<***KEYREMOVED***>"', radio)
processed = mksafe(processed)

output.icecast(%mp3(bitrate=320),
host = "xxx", port = 80,
password = "xxx", mount = "liquidsoap.mp3",
name="Liquidsoap MP3 320kbps",
processed)
```

```
[Liquidsoap_MP3_320kbps:3] Connection setup was successful. 
2019/07/18 13:54:48 [threads:3] Created thread "wallclock_main" (1 total). 
2019/07/18 13:54:48 [clock:4] Main phase starts. 
2019/07/18 13:54:48 [clock.wallclock_main:3] Streaming loop starts, synchronized with wallclock. 
2019/07/18 13:54:48 [mksafe:3] Switch to safe_blank. 
2019/07/18 13:54:48 [safe_blank:4] Activations changed: static=[mksafe:Liquidsoap_MP3_320kbps:Liquidsoap_MP3_320kbps], dynamic=[mksafe:Liquidsoap_MP3_320kbps:Liquidsoap_MP3_320kbps]. 
2019/07/18 13:54:48 [pipe_6130:5]  THIMEO STEREO TOOL 9.34 CONSOLE APPLICATION - (C) HANS VAN ZUTPHEN, 2001-
2019 Registered to Reto Ehrensperger. Creating parameter set objects... 
2019/07/18 13:54:48 [pipe_6130:5]   
2019/07/18 13:54:48 [pipe_6130:5] Converting to using settings from /etc/liquidsoap/rm.sts 
2019/07/18 13:54:48 [pipe_6130:5] Encountered RIFF/WAVE header "fmt ", copying 16 bytes Encountered RIFF/WAVE header "data", processing -268435493 bytes Determining delay... 
2019/07/18 13:54:48 [request.dynamic_6114:4] Remaining: 0.0s, queued: 30.0s, taking: 30.0s 
2019/07/18 13:54:48 [request.dynamic_6114:3] Prepared "/home/autodj/musik/original/5 Seconds Of Summer - Easier (Seeb Remix).mp3" (RID 0). 
2019/07/18 13:54:48 [mksafe:3] Switch to cross_6122 with transition. 
2019/07/18 13:54:48 [safe_blank:4] Activations changed: static=[], dynamic=[mksafe:pipe_6130]. 
2019/07/18 13:54:48 [cross_6122:4] Activations changed: static=[mksafe:pipe_6130], dynamic=[mksafe:pipe_6130]. 
2019/07/18 13:54:48 [pipe_6130:4] Closing process's stdin 
2019/07/18 13:54:48 [pipe_6130:5]  DELAY: 18048 samples (9024 stereo samples) 18014398509219839K input output 
2019/07/18 13:54:48 [pipe_6130:5] 4K #----------------------- #----------------------- 
2019/07/18 13:54:48 [pipe_6130:5] 8K #----------------------- #----------------------- 
2019/07/18 13:54:48 [pipe_6130:5] 12K ------------------------ ------------------------ 
2019/07/18 13:54:48 [pipe_6130:5] 16K #----------------------- #----------------------- 
2019/07/18 13:54:48 [pipe_6130:5] 20K ------------------------ ------------------------ 
2019/07/18 13:54:48 [pipe_6130:5] 24K #----------------------- #----------------------- 
2019/07/18 13:54:48 [pipe_6130:5] 28K ------------------------ ------------------------ 
2019/07/18 13:54:48 [pipe_6130:5] 32K #----------------------- #----------------------- 
2019/07/18 13:54:48 [pipe_6130:5] 36K ------------------------ ------------------------ 
2019/07/18 13:54:48 [pipe_6130:5] 40K #----------------------- #----------------------- 
2019/07/18 13:54:48 [pipe_6130:5] 44K ------------------------ ------------------------ 
2019/07/18 13:54:48 [pipe_6130:5] 48K #----------------------- #----------------------- 
2019/07/18 13:54:48 [pipe_6130:5] 52K ------------------------ ------------------------ 
2019/07/18 13:54:48 [pipe_6130:5] 56K #----------------------- #----------------------- 
2019/07/18 13:54:48 [pipe_6130:5] 60K ------------------------ -----------------------
```

**To Reproduce**
Use Stereo Tool with Pipe as described above.

**Expected behavior**
Audio on Pipe output instead of silence

**Version details**
 - OS: Ubuntu 16.04.6 LTS
 - Version 1.3.7

**Install method**
- opam

**Common issues**
Can't decode or encode in some audio format? Check if you have [installed the correct dependencies](https://www.liquidsoap.info/download.html)

**Comment by smimram** [comment]
Interesting remark from slack so that we do not forget:

I tested with ffmpeg now as well and it does not work either.
This actually should just pick up audio from stdin and forwarding it as wav to stdout:
```
pipe(process='ffmpeg -i - -f wav -')
```

**Comment by RM-FM** [comment]
Are you using Ubuntu 16.04 LTS as well?

**Comment by smimram** [comment]
Debian, but most importantly I am using Liquidsoap 1.4.0 (the latest git version).

**Comment by RM-FM** [comment]
We‘ve already planned to check 1.4 beta as well. Romain mentioned that it will be released soon. Unfortunately we‘ve spotted some issues with the new crossfade as it was not behaving like expected on cue/fade/cross meta data during our latest test on Windows. We will give it a try on Ubuntu next even if I expect that the crossfade will be a desaster :-)

**Comment by RM-FM** [comment]
I can confirm that Stereo Tool is working with Liquidsoap 1.4.0. As expected we are facing some other issues now where we have to take a closer look first.

**Comment by smimram** [comment]
Great, closing the bug then. Feel free to report about your other issues.

**Comment by RM-FM** [comment]
As you might know we have a lot of problems with version 1.4. Beside Stereo Tool everything worked really fine with 1.3.7. Is there any chance to check again why Stereo Tool ist not working using 1.3.x?

**Comment by toots** [comment]
Yes. I can probably backport the 1.4 changes. I'll consider it but I'm not sure how much bugfix we will want to do on the `1.3.x` branch after the initial `1.4.x` release.

Would you mind filling tickets for all your issues? It's great for us to track. So far, I only have the crossfade one.

**Comment by RM-FM** [comment]
Basically it is only crossfading / cue_cut. But there are some other strange behaviors in regards to stability (crossfade applies sometimes/sometimes not, silence on the stream, server restart need in some cases). As we are still in Proof-of-concept mode, it is the wrong point in time for us to focus on a beta version as you will understand surely. Once we‘ve decided to go with Liquidsoap (1.3.7) we are glad to send our input in regards to 1.4 as well :-)

**Comment by toots** [comment]
Ok gotcha. I'll push a fix to `1.3.7` soon. Your input would be very helpful to release a solid `1.4.0` next.

**Comment by RM-FM** [comment]
Great! Looking forward... We‘ve already set up two environments - one for 1.3.7 and the other one for Beta testing.

**Comment by RM-FM** [comment]
Boom! Good news... Stereo Tool is fully working under 1.3.7 using Ubuntu 18.04 :-)

**Comment by RM-FM** [comment]
Finally I can confirm that the issue is related to meta data only. Using Pipe incl. Meta Data in 1.3.7 will cause Stereo Tool to fail.

**Comment by toots** [comment]
Ok cool. Let's circle back on slack to I get keep track of the issues we can fix on `1.4.0`. I have a branch with the delay metadata injection in `pipe`, seems to be working really good.
